api.lua

```diff
@@ -95,36 +95,55 @@ function areas:getAreasIntersectingArea(pos1, pos2)
 	return res
 end
 
--- Checks if the area is unprotected or owned by you
+-- Returns smallest area at position and its id or nil.
+-- If multiple areas have the same volume, larger id takes precedence.
+function areas:getSmallestAreaAtPos(pos)
+	local smallest_area, smallest_id, smallest_volume, volume = nil
+	for id, area in pairs(self:getAreasAtPos(pos)) do
+		volume =	(area.pos2.x - area.pos1.x + 1)
+					* (area.pos2.y - area.pos1.y + 1)
+					* (area.pos2.z - area.pos1.z + 1)
+		if not smallest_volume or smallest_volume >= volume then
+			smallest_area = area
+			smallest_id = id
```

Comment:
Not needed to keep track of. The ID can be retrieved from `area.id`. Thus, only one return value is necessary.

Reply:
Negative. ``print(dump(area))`` produces
```lua
{
	faction_open = {
		"z"
	},
	pos2 = {
		x = 607,
		y = 22,
		z = 1916
	},
	name = "big",
	owner = "singleplayer",
	pos1 = {
		x = 601,
		y = 13,
		z = 1910
	}
}
```
no ``id`` field.

Reply:
Oh I see. I got mislead by the search match in `hud.lua`, `areas:getExternalHudEntries(pos)` where an `id` field may be present.
Returning the area table and ID does make sense then.